parser_library/test/processing/aread_test.cpp

```diff
@@ -359,3 +359,385 @@ Line5
     ASSERT_EQ(diags.size(), 1);
     EXPECT_EQ(diags[0].diag_range.start.line, 15);
 }
+
+namespace {
+class asm_options_invalid_lib_provider : public parse_lib_provider
+{
+    std::unordered_map<std::string, std::string> m_files;
+
+public:
+    parse_result parse_library(const std::string& library, analyzing_context ctx, library_data data) override
+    {
+        auto it = m_files.find(library);
+        if (it == m_files.end())
+            return false;
+
+        auto a = std::make_unique<analyzer>(it->second, analyzer_options { library, this, std::move(ctx), data });
+        a->analyze();
+        a->collect_diags();
```

Comment:
```suggestion
```

Reply:
We end up doing this step during normal processing, so while it is redundant here, I'd prefer keeping it.